src/imio/esign/browser/actions.py

```
@@ -121,3 +130,65 @@ def available(self):
+                lines.append(u"{}{}: {},".format(inner, key, self._render_value(v, inner)))
+            lines.append(u"{}}}".format(indent))
+            return u"\n".join(lines)
+        elif isinstance(value, (list, tuple)):
+            if not value:
+                return u"[]"
+            lines = [u"["]
+            for item in value:
+                lines.append(u"{}{},".format(inner, self._render_value(item, inner)))
+            lines.append(u"{}]".format(indent))
+            return u"\n".join(lines)
+        elif isinstance(value, basestring) and re.match(r"^[0-9a-f]{32}$", value):
```

Ruff (0.15.4)

[error] Undefined name `basestring`

(F821)